ch3/recursion.js

```diff
@@ -0,0 +1,32 @@
+const assert =  require('assert');
+/**
+ * checks if input value is even
+ * @param {*} n - input value
+ * @returns true if even, false if not
+ */
+const isEven = (n) => {
+  n = Math.abs(n);
+
+  //these are so short. Can i just keep them one line? Would anyone really care?
+  if (n === 0) return true; 
+  else if (n === 1) return false;
+  return isEven(n-2);
+  //which one? first one easier to read
+  /*return (n === 0) ? true :
+  (n === 1) ? false :
+  isEven(n - 2);*/
+}
+
+//Is there harm in using assrt or jest in this case?
+//I can see how it's a time suck but if you have
+//many function calls, you can get lost on which one is failing
+//and since it's a recursive call, the input value changes from your original
+//input each time
```

Comment:
no harm, candidates normally use `assert` instead of `jest` since `jest` requires a bit more scaffolding. if you find yourself in an interview where you can't import `'assert'` for whatever reason, you can also do something like:
```js
[
 [2, true],
 [1, false],
 [-1, false]
].forEach(([input, expectedOutput]) => `For input ${input}, expected ${expectedOutput}, received ${isEven(input)}`);
```

Reply:
Do I need to chain this with assert? Or it prints on its own?

Reply:
oh i see. I was confused because there was no console.log in this. I got it now.